src/components/Attachments/AttachmentCarousel/index.js

```diff
@@ -153,6 +168,10 @@ function AttachmentCarousel({report, reportActions, source, onNavigate, setDownl
         [activeSource, canUseTouchScreen, setShouldShowArrows, shouldShowArrows],
     );
 
+    if (isLoadingReport || isLoadingReportAction) {
+        return <FullScreenLoadingIndicator />;
+    }
+
```

Comment:
We usually avoid duplicated code. What do you think about adding this logic in parent component - AttachmentModal?